docs/usage/serve.rst

```diff
@@ -39,3 +39,32 @@ locations like ``/etc/environment`` or in the forced command itself (example bel
     has been introduced in v7.2. We recommend to use
     ``no-port-forwarding,no-X11-forwarding,no-pty,no-agent-forwarding,no-user-rc``
     in this case.
+    
+SSH Configuration
+~~~~~~~~~~~~~~~~~
+``borg serve``'s pipes (``stdin``/``stdout``/``stderr``) are connected to the ``sshd`` process on the server side. In the event that the SSH connection between ``borg serve`` and the client is disconnected abnormally (for example, due to a network outage), it can take a long time for ``sshd`` to notice the client is disconnected. In the meantime, ``sshd`` continues running, and as a result so does the ``borg serve`` process holding the lock on the repository. This can cause subsequent ``borg`` operations on the remote repository to fail with the error: ``Failed to create/acquire the lock``.
+
+In order to avoid this, it is recommended to perform the following additional SSH configuration:
+
+In the client side's ``~/.ssh/config`` file:
+::
+
+    Host backupserver
+            ServerAliveInterval 10
+            ServerAliveCountMax 30
+
+Replacing ``backupserver`` with the hostname or FQDN of of the borg server.
+
+This will cause the client to send a keepalive to the server every 10 seconds. If 30 consecutive keepalives are sent without a response (a time of 300 seconds), the connection will be terminated, causing the borg process to terminate gracefully.
+
+On the server side's ``sshd`` configuration file (typically ``/etc/ssh/sshd_config``):
+::
+
+    ClientAliveInterval 10
+    ClientAliveCountMax 30
+
+This will cause the server to send a keep alive to the client every 10 seconds. If 30 consecutive keepalives are sent without a response (a time of 300 seconds), the connection will be terminated, causing the ``borg serve`` process to terminate gracefully and release the lock on the repository.
+
+If you then run borg commands with ``--lock-wait 600``, this gives sufficient time for the borg serve processes to terminate after the SSH connection is torn down after the 300 second wait for the keepalives to fail.
+
+You may, of course, modify the timeout values demonstrated above to values that suit your environment and use case.
```

Comment:
guess that is somehow clear, the only question remaining is what the safe minimum would be.

Reply:
Perhaps I should change the wording to:

```If you then run borg commands with ``--lock-wait 600``, this should give sufficient time for ssh and borg processes to gracefully terminate in the event processes are left running due to a network issue.```

Reply:
With regards to a safe minimum, I've found that borg serve needs a little bit of time to "tidy up" once the connection fails. In my environment, I simply doubled it and this works for me, however I'm not sure what a safe minimum would be. What internal tasks is borg serve doing when terminating on connection loss, and what is a safe maximum on how long should these take?

Reply:
maybe i phrased it badly. i didn't mean to reduce lock-wait, but i also meant the 300s here.

lock-wait is just the maximum waiting time. if everything works fine, the lock will go away and it will wait less than that.